### ISSUE TITLE
erroneous pipeline: no element "vp8enc"

### ISSUE THREAD

**Issue by manumenon** [openvinotoolkit/openvino#1229]
my pipeline :
gst-launch-1.0
videotestsrc !
video/x-raw,width=640,height=480,framerate=30/1 !
videoscale ! videorate ! videoconvert ! timeoverlay !
vp8enc error-resilient=1 !
rtpvp8pay ! udpsink host=192.168.1.47 port=10027

not able to find a solution
**erroneous pipeline: no element "vp8enc"**

i have **intel-openvino-GStreamer-rt-ubuntu-bionic-2020.3.194**,  intel vino dev is installed. i am running this in intel nuc. t want to transcode to rtp (vp8) and puts it to Janus. I need to use openvino for processing each frame also

**Comment by manumenon** [comment]
I have installed this gstreamer-good-plugins (https://gstreamer.freedesktop.org/documentation/vpx/vp8enc.html?gi-language=c) and done some debugging but even then it's showing the same. I feel since the openvino is installing some custom paths for gst plugins this is not getting loaded.
![image](https://user-images.githubusercontent.com/25078680/86598117-4aeba180-bfba-11ea-883a-2a2933fe7067.png)
This is what i mean. not sure it is taking the plugins which I installed. I need to use openvino as well as GStreamer

**Comment by brmarkus** [comment]
On my Ubuntu-based Linux dev machine I can do a "gst-inspect vp8enc", which lists me

```
...
Plugin Details:
  Name                     vpx
  Description              VP8 plugin
  Filename                 /usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstvpx.so
  Version                  1.8.3
  License                  LGPL
  Source module            gst-plugins-good
  Source release date      2016-08-19
  Binary package           GStreamer Good Plugins (Ubuntu)
  Origin URL               https://launchpad.net/distros/ubuntu/+source/gst-plugins-good1.0
...
```

Can you check whether you have the `vpx` plugin installed and have a library called `libgstvpx.so` on your system?

**Comment by manumenon** [comment]
> On my Ubuntu-based Linux dev machine I can do a "gst-inspect vp8enc", which lists me
> 
> ```
> ...
> Plugin Details:
>   Name                     vpx
>   Description              VP8 plugin
>   Filename                 /usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstvpx.so
>   Version                  1.8.3
>   License                  LGPL
>   Source module            gst-plugins-good
>   Source release date      2016-08-19
>   Binary package           GStreamer Good Plugins (Ubuntu)
>   Origin URL               https://launchpad.net/distros/ubuntu/+source/gst-plugins-good1.0
> ...
> ```
> 
> Can you check whether you have the `vpx` plugin installed and have a library called `libgstvpx.so` on your system?
![image](https://user-images.githubusercontent.com/25078680/86600705-0d891300-bfbe-11ea-9b67-8bf41f4eab4d.png)

i can see this file .

**Comment by brmarkus** [comment]
Can you manually first try what is mentioned here, `https://stackoverflow.com/a/59919316`, please?

Like (using your path)

`gst-inspect-1.0 vp8enc --gst-plugins-path=/usr/lib/x86_64-linux-gnu/gstreamer-1.0/`

Not sure the ENV variable `GST_PLUGIN_PATH` accepts multiple paths&folders...

**Comment by brmarkus** [comment]
From `https://developer.gnome.org/gstreamer/stable/gst-running.html`:

> GST_PLUGIN_PATH, GST_PLUGIN_PATH_1_0.  This environment variable can be set to a colon-separated list of paths. GStreamer will scan these paths for GStreamer plug-ins. These plug-ins will be loaded in addition to, and before, the plug-ins in the system paths.

Sounds good, can you try to expand the variable from the OpenVINO-GVA setup script?

Like

`export GST_PLUGIN_PATH=$(GST_PLUGIN_PATH):/usr/lib/x86_64-linux-gnu/gstreamer-1.0/`

**Comment by iefode** [comment]
@adranit Could you please help?

**Comment by adranit** [comment]
Hi @manumenon,
We'll investigate why vp* elements are not included in the package. 
In the meantime, I can recommend expanding the GST_PLUGIN_PATH variable, as @brmarkus wrote earlier:

export GST_PLUGIN_PATH=$(GST_PLUGIN_PATH):/usr/lib/x86_64-linux-gnu/gstreamer-1.0/

**Comment by manumenon** [comment]
> Hi @manumenon,
> We'll investigate why vp* elements are not included in the package.
> In the meantime, I can recommend expanding the GST_PLUGIN_PATH variable, as @brmarkus wrote earlier:
> 
> export GST_PLUGIN_PATH=$(GST_PLUGIN_PATH):/usr/lib/x86_64-linux-gnu/gstreamer-1.0/

This seems to work. i will keep updating

**Comment by manumenon** [comment]
> From `https://developer.gnome.org/gstreamer/stable/gst-running.html`:
> 
> > GST_PLUGIN_PATH, GST_PLUGIN_PATH_1_0.  This environment variable can be set to a colon-separated list of paths. GStreamer will scan these paths for GStreamer plug-ins. These plug-ins will be loaded in addition to, and before, the plug-ins in the system paths.
> 
> Sounds good, can you try to expand the variable from the OpenVINO-GVA setup script?
> 
> Like
> 
> `export GST_PLUGIN_PATH=$(GST_PLUGIN_PATH):/usr/lib/x86_64-linux-gnu/gstreamer-1.0/`

yes . This is working. if i export a path it seems to work

**Comment by adranit** [comment]
We have included vp8 plugin into 2021.1 release.
